src/code/z_play.c

```diff
@@ -590,175 +612,184 @@ void Gameplay_Update(GlobalContext* globalCtx) {
                         } else {
                             globalCtx->transitionCtx.destroy(&globalCtx->transitionCtx);
                             func_800BC88C(globalCtx);
-                            globalCtx->transitionMode = 0;
+                            globalCtx->transitionMode = TRANS_MODE_OFF;
+
                             if (gTrnsnUnkState == 3) {
                                 TransitionUnk_Destroy(&sTrnsnUnk);
                                 gTrnsnUnkState = 0;
                                 R_UPDATE_RATE = 3;
                             }
                         }
-                        globalCtx->sceneLoadFlag = 0;
+
+                        globalCtx->transitionTrigger = TRANS_TRIGGER_OFF;
                     } else {
                         globalCtx->transitionCtx.update(&globalCtx->transitionCtx.data, R_UPDATE_RATE);
                     }
                     break;
             }
 
+            // update non-instance type transitions
             switch (globalCtx->transitionMode) {
-                case 4:
-                    D_801614C8 = 0;
+                case TRANS_MODE_FILL_WHITE_INIT:
+                    sTransitionFillTimer = 0;
                     globalCtx->envCtx.fillScreen = true;
                     globalCtx->envCtx.screenFillColor[0] = 160;
                     globalCtx->envCtx.screenFillColor[1] = 160;
                     globalCtx->envCtx.screenFillColor[2] = 160;
-                    if (globalCtx->sceneLoadFlag != -0x14) {
+
+                    if (globalCtx->transitionTrigger != TRANS_TRIGGER_OUT) {
                         globalCtx->envCtx.screenFillColor[3] = 0;
-                        globalCtx->transitionMode = 5;
+                        globalCtx->transitionMode = TRANS_MODE_FILL_IN;
                     } else {
                         globalCtx->envCtx.screenFillColor[3] = 255;
-                        globalCtx->transitionMode = 6;
+                        globalCtx->transitionMode = TRANS_MODE_FILL_OUT;
                     }
                     break;
 
-                case 5:
-                    globalCtx->envCtx.screenFillColor[3] = (D_801614C8 / 20.0f) * 255.0f;
-                    if (D_801614C8 >= 20 && 1) {
-                        globalCtx->state.running = 0;
+                case TRANS_MODE_FILL_IN:
+                    globalCtx->envCtx.screenFillColor[3] = (sTransitionFillTimer / 20.0f) * 255.0f;
+
+                    if (sTransitionFillTimer >= 20 && 1) {
```

Comment:
```suggestion
                    if (sTransitionFillTimer >= 20 && true) {
```
Seems like a strange if-condition. The `true` is needed?